src/query/functions/src/scalars/array.rs

```diff
@@ -170,6 +170,48 @@ pub fn register(registry: &mut FunctionRegistry) {
         ),
     );
 
+    registry.register_2_arg_core::<NullableType<EmptyArrayType>, NullableType<EmptyArrayType>, EmptyArrayType, _, _>(
+        "concat",
+        FunctionProperty::default(),
+        |_, _| FunctionDomain::Full,
+        |_, _, _| Value::Scalar(()),
+    );
+
+    registry.register_combine_nullable_2_arg::<EmptyArrayType, ArrayType<NullableType<GenericType<0>>>, ArrayType<NullableType<GenericType<0>>>, _, _>(
```

Comment:
could use `register_passthrough_nullable_2_arg `, because no new `NULL` will be generated here.
`ArrayType<GenericType<0>>` is ok.